androidApp/src/main/java/press/home/HomeView.kt

```diff
@@ -98,7 +107,9 @@ class HomeView @AssistedInject constructor(
     )
   }
 
-  private val noteEditorPage = ExpandablePageLayout(context).apply {
+  // Clarify the type here to clear the 'type checking recursive problem'.
+  private val noteEditorPage: ExpandablePageLayout = ExpandablePageLayout(context).apply {
+    doOnNextCollapse { noteForEditor = null }
```

Comment:
Could this be moved to the `observable` delegate where you're calling `notesList.collapse()` when the note is null?

Reply:
@saket It was there in my first implement this. But Android Studio warns me about a 'type checking recursive problem'. I guess it thinks calling setter inside a setter is bad. Let me see it again later.

Reply:
Right right, but the comment isn't needed. If someone tries to remove the explicit return type, the IDE will complain anyway.